### ISSUE TITLE
Intellisense does not show signature for nested function call

### ISSUE THREAD

**Issue by rhys-vdw** [dotnet/vscode-csharp#6670]
Type: <b>Bug</b>

## Issue Description ##

When entering the parentheses for a nested function call, the intellisense popup will not show the parameters required.

## Steps to Reproduce ##

Use something like this:
```cs
class Foo {
  public Foo(int showMe) {}

  public static void Do(Foo foo) {
    Do(new Foo( // <-- cursor here
  }
}
```
It won't show the `showMe` parameter in the popup.

## Expected Behavior ##

See parameters for nested function.

Like this, you can see `showMe`:
![image](https://github.com/dotnet/vscode-csharp/assets/822105/2fc99860-b6d6-46aa-9d03-bdd2f38736d8)

## Actual Behavior ##

See parameters for outer function.

![image](https://github.com/dotnet/vscode-csharp/assets/822105/46e5d65b-f409-46fd-99db-5ad2d3e9740a)

### C# log ###
<details>Post the output from Output-->C# here</details>

### C# LSP Trace Logs ###
<details>Post the output from Output-->C# LSP Trace Logs here.  Requires `dotnet.server.trace` to be set to `Trace`</details>

## Environment information ##

**VSCode version**: 1.84.2
**C# Extension**: 2.10.28
**Using OmniSharp**: false

<details><summary>Dotnet Information</summary>

Host:
  Version:      7.0.13
  Architecture: x64
  Commit:       3f73a2f186

.NET SDKs installed:
  No SDKs were found.

.NET runtimes installed:
  Microsoft.NETCore.App 7.0.13 [c:\Users\rhysv\AppData\Roaming\Code\User\globalStorage\ms-dotnettools.vscode-dotnet-runtime\.dotnet\7.0.13~x64\shared\Microsoft.NETCore.App]

Other architectures found:
  x86   [C:\Program Files (x86)\dotnet]

Environment variables:
  Not set

global.json file:
  Not found

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download
</details>
<details><summary>Visual Studio Code Extensions</summary>

|Extension|Author|Version|Folder Name|
|---|---|---|---|
|change-case|wmaurer|1.0.0|wmaurer.change-case-1.0.0|
|copilot|GitHub|1.135.0|github.copilot-1.135.0|
|copilot-chat|GitHub|0.10.1|github.copilot-chat-0.10.1|
|csdevkit|ms-dotnettools|1.1.9|ms-dotnettools.csdevkit-1.1.9-win32-x64|
|csharp|ms-dotnettools|2.10.28|ms-dotnettools.csharp-2.10.28-win32-x64|
|docomment|k--kato|1.0.0|k--kato.docomment-1.0.0|
|EditorConfig|EditorConfig|0.16.4|editorconfig.editorconfig-0.16.4|
|emoji-code|idleberg|0.12.1|idleberg.emoji-code-0.12.1|
|font-switcher|evan-buss|4.1.0|evan-buss.font-switcher-4.1.0|
|gitignore|codezombiech|0.9.0|codezombiech.gitignore-0.9.0|
|gitlens|eamodio|14.4.1|eamodio.gitlens-14.4.1|
|guid|kosunix|0.0.2|kosunix.guid-0.0.2|
|handlebars-formatter|mfeckies|1.1.0|mfeckies.handlebars-formatter-1.1.0|
|hot-dog-stand|somekittens|6.6.9|somekittens.hot-dog-stand-6.6.9|
|ink|bruno-dias|1.3.1|bruno-dias.ink-1.3.1|
|keyoti-changeallendoflinesequence|vs-publisher-1448185|0.0.3|vs-publisher-1448185.keyoti-changeallendoflinesequence-0.0.3|
|lorem-ipsum|Tyriar|1.3.1|tyriar.lorem-ipsum-1.3.1|
|monokai-charcoal-high-contrast|74th|3.4.0|74th.monokai-charcoal-high-contrast-3.4.0|
|prettier-vscode|esbenp|10.1.0|esbenp.prettier-vscode-10.1.0|
|sort-lines|Tyriar|1.11.0|tyriar.sort-lines-1.11.0|
|theme-cotton-candy|forgng|3.0.0|forgng.theme-cotton-candy-3.0.0|
|toggle-hidden|adrianwilczynski|1.0.2|adrianwilczynski.toggle-hidden-1.0.2|
|vdf|GEEKiDoS|0.0.7|geekidos.vdf-0.0.7|
|vim|vscodevim|1.26.2|vscodevim.vim-1.26.2|
|vscode-autohotkey-plus|cweijan|2.7.1|cweijan.vscode-autohotkey-plus-2.7.1|
|vscode-counter|uctakeoff|3.2.2|uctakeoff.vscode-counter-3.2.2|
|vscode-dotnet-runtime|ms-dotnettools|2.0.0|ms-dotnettools.vscode-dotnet-runtime-2.0.0|
|vscode-templates|redxam|1.1.0|redxam.vscode-templates-1.1.0|
|vscode-unitymeta|PTD|0.0.7|ptd.vscode-unitymeta-0.0.7|
|vscodeintellicode-csharp|ms-dotnettools|0.1.26|ms-dotnettools.vscodeintellicode-csharp-0.1.26-win32-x64|
|vstuc|visualstudiotoolsforunity|0.9.3|visualstudiotoolsforunity.vstuc-0.9.3|
|wordcount|ms-vscode|0.1.0|ms-vscode.wordcount-0.1.0|;

</details>

Extension version: 2.10.28
VS Code version: Code 1.84.2 (1a5daa3a0231a0fbba4f14db7ec463cf99d7768e, 2023-11-09T10:51:52.184Z)
OS version: Windows_NT x64 10.0.22621
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|AMD Ryzen 5 5600X 6-Core Processor              (12 x 3700)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|undefined|
|Memory (System)|31.93GB (5.71GB free)|
|Process Argv|--crash-reporter-id 9635b07e-a69c-4bcd-ae99-dc8336779313|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
vslsvsres303:30308271
vserr242:30382549
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
vscorecescf:30445987
vscod805cf:30301675
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593:30376534
pythonvs932:30410667
py29gd2263:30880072
vsclangdc:30486549
c4g48928:30535728
dsvsc012:30540252
pynewext54:30695312
azure-dev_surveyone:30548225
2e4cg342:30602488
89544117:30613380
2i9eh265:30646982
showlangstatbar:30737416
pythonfmttext:30731395
fixshowwlkth:30771522
showindicator:30805244
pythongtdpath:30769146
i26e3531:30792625
welcomedialogc:30887144
pythonnosmt12:30797651
pythonidxpt:30866567
pythonnoceb:30805159
synctok:30869157
dsvsc013:30795093
dsvsc014:30804076
dsvsc015:30845448
pythontestfixtcf:30871695
pythonregdiag2:30871582
pyreplss2:30886141
pythonmypyd1:30879173
pythoncet0:30885854
pythontbext0:30879054
dsvsc016cf:30886111
dsvsc017cf:30886113
dsvsc018cf:30886115
aa_t_chat:30882232

```

</details>

**Comment by dibarbet** [comment]
Looks like even if the old signature help is dismissed it will never show the signature help for the nested call
![nested_signature_help](https://github.com/dotnet/vscode-csharp/assets/5749229/3b530ce1-abf1-4915-b2e4-d109b27fb486)

Almost certainly a server side bug.

**Comment by rhys-vdw** [comment]
Does this being a server bug mean that it needs to be reported elsewhere? It's quite a bothersome issue.

**Comment by dibarbet** [comment]
This fix is in the 2.33.17 release
